aea/helpers/ipfs/base.py

```diff
@@ -157,20 +291,10 @@ def _generate_multihash(cls, pb_data: bytes) -> str:
         """
         multihash_bytes = cls._generate_multihash_bytes(pb_data)
         ipfs_hash = base58.b58encode(multihash_bytes)
-        return str(ipfs_hash, "utf-8")
```

Comment:
I'm not exactly sure why but using utf-8 encoding was giving out wrong hashes so I switched to str.decode() method instead. Should I investigate the reason?

Reply:
that makes no sense to me, as the default encoding by `bytes.decode` is utf-8 (with errors='strict')

Reply:
Please investigate

Reply:
So michiel was right, using errors='strict' did solve the problem for some of the files, but it turned out there was also a problem with encoding which I accidentally solved while writing code to support wrapper nodes.